### ISSUE TITLE
We may have to consider to change the third-part icons to it's defaults

### ISSUE THREAD

**Issue by Diolinux** [linuxmint/mint-y-icons#241]
Hi, that's a minor change but impacts a lot the credibility of Linux Mint for the professional market. How would you feel if a vendor changes the brand of your company or product in order to ship one very different instead?

I was discussing this with Clement, and he suggests that I came here to debate this idea.

The Linux Mint icon theme is beauty (IMO), but is getting outdated fast, and the main reason for that (again IMO), is especially the third-part icons, like Steam, Firefox, GIMP, well... and a bunch of
others too, Nvidia, OBS Studio etc.

Look at Canonical's guideline: https://samuelhewitt.com/ubuntu-icon-design-guide

The shape of the icons is not a problem, the problem is changing the brand of others, this shouldn't happen.

Spotify is one (bad) example, as you can see: https://slack-files.com/TL04QTEA1-FRJR1PHR7-bf21e51ae5

Here (up), I'd suggest a version that maintains the shape of Linux Mint's theme but use the icon from the original developer, a similar approach to what Apple does.

As I said, Spotify is a bad example, take GIMP or Firefox, the actual icons don't even look like the original art created by the projects. Use the actual icons of application will reduce the time and effort from the design team of Linux Mint, let them focus on the details and will bring a sensation of familiarity, and this is important for news users.

I think we want the users exploring the software center reach apps thinking 'oh, i know this software", and the icon is a great part of this experience, what we don't want to see is users guessing "what software is this" and beyond, we don't want to Mint get problems with the law if laptops for desktop will be shipping with brands corrupted. As I said, it's an important thing for the market.

There is a reason to GNOME and KDE don't change the brands, and just customize their own icons.

I hope this helps some way.

All the best.

**Comment by raulcraveiro** [comment]
I think that the current icon theme could be an option inside Linux Mint, but not by default. It's not cool to change other brands logos, since there's a LOT of study to came with a final logo. 

Like @Diolinux said, I think that the iOS icons are a good example of how to keep the rounded square shape, but respecting the brand's logo.

**Comment by leonardfreitas** [comment]
I believe @Diolinux is right when it comes to not changing company brands, it can help users and businesses. I don't know if there's an orientation to this, but it's a great idea. let's talk more about it

**Comment by Klauswk** [comment]
This is one of the few things i really dislike about mint, everytime i install on some machine i end up deleting the icons inside mint-y folder to match the original.

I fully support this change and agree with @Diolinux

**Comment by HenriqueAD7** [comment]
Keeping the original icons, as well as keeping the brand recognizable to the user, can prevent future problems in case of selling some equipment with Linux Mint.  In particular, and with people I know and use distro, icons are always customized because of this "problem".  Examples like Yaru are interesting.  Modern icons and holds the third party icons.

**Comment by clefebvre** [comment]
First, let's clarify a few things we can all agree on.

1. It's the purpose of an icon theme to make icons look consistent. If every icon in your window list had a completely different style, they'd be no point in having a theme. That's its main purpose.

2. Each individual application has the choice whether to use icon themes or not. Most apps do, they refer to an icon name and thus look different in different icon themes. Some don't, they specifically point to an image they ship, or use unique IDs (Electron Apps do that for instance).

3. Users have the choice to choose the icon theme they want. System Settings -> Themes -> Adwaita, no more spotify icon in the theme.

4. Themed icons don't just adhere to a style dictated by the theme, they also portray the identity of the app and make it recognizable. In mint-y, spotify, gimp and firefox don't look exactly like their upstream counterparts but there's no ambiguity as to what app they symbolize.

5. It's common practice not only in Linux, but in smartphones and on TV to theme artwork to give it a unify look. In fact, you almost have to do it if you want things to look professional.

6. We can care about upstream brands wanting to look the same in Linux and Windows, but our primary concern should be the users. If something looks better and gives a better user experience that's what we should aim for as a priority.

I hope we agree on these as a start.

**Comment by clefebvre** [comment]
Now, from a technical point of view, I'd like to go through how we "could" (not saying we should) make it so people can opt in/out branded app icons.

1. Some themes include app icons, some don't, some include a few, others include many. The user chooses the theme he/she wants, and so alongside with that already has the possibility to opt for a theme which doesn't include app icons. We can argue whether or not Linux Mint should or shouldn't "by default", but when it comes to the user's ability to choose, it's already there. That part of the problem is solved.

2. A theme can inherit another theme, so in theory mint-y-apps could inherit mint-y and mint-y could ship without app icons. This is a mess though because of the already existing dark and light variations which are there to facilitate apps which don't use symbolic icons to look OK, and the many color variations... we already use inheritance a lot, so that would be a complete mess, not to mention point 1 already deals with the problem.

3. If the idea isn't to remove app icons, but to include untouched brand icons (i.e. not-stylized) inside a common icon shape, then we can make a theme which inherits mint-y and overrides these icons. This would be a community project though, cause as illustrated in point 2 it couldn't work with dark vs light and the color variations without being super messy.

4. A command can be provided to remove app icons or to switch them to a non-stylized form, we could even provide a UI option to choose between stylized and non-stylized. That's merely a workaround to please people, not a real solution. I just mention it here as I'm going through the different technical approaches. The solution is obviously to talk about all of this, get passed any consideration that isn't important to us and define what's best going forward, not to add yet another little configuration option very few people might use.

5. App icons can be shipped as part of a different package. That can work, but again it's a workaround.. it just makes it easier for people to choose, and they already can.

The "user choice" rationale is not interesting. They already have a choice and even if they decided to use a theme which doesn't do what they want, they have the possibility to hack it (remove files) or to properly inherit it (define a super-theme which inherits it). We can make this easier of course, but we're talking niche feature here. If this isn't niche, then we're much better off focusing on a larger picture of debating this in terms of defaults.

In my first comment I also illustrated the fact that Windows wasn't relevant (it has no theme for apps to use, so how could they?) and that our user experience is more important than brand considerations.

With all that in mind, I'd say let's focus on what looks best for everybody, without caring about theme-less OSes (Windows), and arguments which aren't relevant (user ability to choose, brand right not to be stylized..etc).

Questions like the ones below are more relevant imo:

- User experience: Does styling app icons make the apps not recognizable? (i.e. are people confused about this being spotify, gimp, firefox?)
- Taste: Do people enjoy consistent styles and theming in general? or would they rather see the exact brand logos they're used to?
- How far do we go: If we're to stop styling app icons, do we just paste the logo on top of a shape? or do we just not provide an icon at all? For spotify for instance, this would be the current icon vs the spotify logo inside a mint-y shape vs the spotify logo on its own.
- What's the scope of it: If we do this for spotify, gimp and firefox... then we do steam and virtualbox... then we do filezilla and libreoffice... all the way down to any apps really? Some apps ship with awful looking icons... and some adhere to completely different looking styles. How far do we go with this? We just stick to the few apps everybody knows outside of Linux or not? and if so, why?

**Comment by clefebvre** [comment]
Here's a sports competition league table:

![image](https://user-images.githubusercontent.com/1138515/72425739-428b7a00-3780-11ea-8e68-3b7aaf8f6ec0.png)

Now think of our spotify icon. 

![image](https://user-images.githubusercontent.com/1138515/72425872-84b4bb80-3780-11ea-9a52-b1216505fc4f.png)

People on TV are theming the stuff they show. We see that in every big tournament, flags aren't the flags people know, they're stylized versions of them.

Flags don't have the same size, in fact they don't have the same shape even. Some are 4/3, some are square etc.. what you see here are symbolized icons. They take the most recognizable aspects (i.e. their core identity) of the flags and apply that to a shape, a level of detail, and sometimes even a color palette which is consistent across the theme.

What they do here is exactly what icon themes do basically.

**Comment by clefebvre** [comment]
Here are a few examples: 

![image](https://user-images.githubusercontent.com/1138515/72426745-37d1e480-3782-11ea-9c4e-e93ec49f14e1.png)

![image](https://user-images.githubusercontent.com/1138515/72426985-baf33a80-3782-11ea-975b-8d08096c9659.png)

![image](https://user-images.githubusercontent.com/1138515/72427050-d8280900-3782-11ea-8de5-5915b29ab2e6.png)

It's common practice on TV, in magazines, pretty much anywhere really. I'm not saying it's done all the time, but it's definitely done enough that people are used to it.

**Comment by clefebvre** [comment]
I almost forgot, one really widespread practice:

![image](https://user-images.githubusercontent.com/1138515/72427627-1245da80-3784-11ea-8de6-9c6c617c7843.png)

**Comment by Diolinux** [comment]
I agree with you Clem, consistency is very important, and the theme helps the visual look coherent. But, coherency doesn't come from shapes necessarily, but from visual consistency. Look at the icons on a dock of macOS Catalina for example or the side-bar from Ubuntu.

![Apple-previews-macOS-Catalina-Gallery-screen-06032019_big_carousel jpg large](https://user-images.githubusercontent.com/13944967/72435912-fe977580-377d-11ea-9874-dec6e7d41f61.jpg)


The icon from Spotify on Linux Mint is the less problematic, cause is similar to the original, as well the chrome. But others like GIMP, Firefox, Lutris, etc. change the original colors and design of the originals. In your last example, notice, the twitter icon for example, is just a color variante of a well knowed brand, just like we have now with the new Linux Mint's logo.

What Mint-Y does to third-brands is like transform the twitter logo into this: 

![screenshot_29](https://user-images.githubusercontent.com/13944967/72436010-369eb880-377e-11ea-9dd4-746a26cf30da.png)


No, add a dark variant or just set a rounded shape. The flatpak softwares on the store don't respect the Mint-Y icons anyway, so you probably will end up with different icons, and they look beautiful.

![screenshot_28](https://user-images.githubusercontent.com/13944967/72435947-1242dc00-377e-11ea-97f1-142f905cd69f.png)

See the problem:

![screenshot_30](https://user-images.githubusercontent.com/13944967/72436167-9006e780-377e-11ea-8276-32264a1126e5.png)

Customization is Core for the Linux community, no doubt, and yes, we can change all we want on the OS, and this is great but is a good major with other developers to ship their very original artwork, isn't it?

Samsung used to do a similar thing with the icons of Galaxy Line, but notice how they manage the icons from third-part:

![630px-NorthernTouch_Home-Screen-Galaxy-S8-03](https://user-images.githubusercontent.com/13944967/72436613-9fd2fb80-377f-11ea-9a23-8fca62706149.png)

As we can see, some of them don't go well with the rounded shape, but still, it maintains the original art, avoiding distortion, no color changing, no refactored design.

Another example from the original Android icons on my phone:

![photo_2020-01-15_10-16-11](https://user-images.githubusercontent.com/13944967/72436920-5800a400-3780-11ea-88be-331e43011f5c.jpg)


All rounded as well, but look at icons from Twitch, YouTube, the camera icon, etc. When the background color can fulfill the background of the icon, we have a perfect circle, but sometimes, if the color fulfills the background, we lose information about the icon itself, that's the case of YouTube. Imagine it all red with the "play" button painted in white. The shape of Youtube's logo doesn't appear, and that's a problem. (or would be)

Mint-y has some icons who don't even come close to the original one, show to a person only the icon, without the name and ask him/her to guess what app is:

![screenshot_32](https://user-images.githubusercontent.com/13944967/72437203-fdb41300-3780-11ea-837d-2cdf6b5283da.png)

That's no VLC, what happen with the iconic traffic cone?

Or Wine, this is not the logo, not even close:

![screenshot_31](https://user-images.githubusercontent.com/13944967/72437298-32c06580-3781-11ea-93a3-9b0acd0e6759.png)

In the end of the day, the user will change to whatever he likes, but, use an icon theme who cover all the application is bad for the project for these reasons:

- You will never be able to cover EVERY app on Linux, so, when the user installs something different, it will have inconsistency.

Ex:

![screenshot_33](https://user-images.githubusercontent.com/13944967/72438214-1e7d6800-3783-11ea-9618-446e4a54936d.png)


That's my Linux Mint, I'm currently using Linux Mint's Mint-Y-Grey-Dark, and I already have icons that don't feet. like Piper (right next to Steam), and Lutris (next to the Piper, which don't have the logo of Lutris), not to mention the others, like slack, Twitch, Simplenote, Skype, etc.

- Change the icon theme to refresh it, from time to time, it's also hard, there are too many icons to work on it.

- As a consequence of the two above, the theme of Linux Mint can be outdated faster than usual, and nowadays we already have icons who do not respect Mint-Y. If the app icons aren't made by the Mint-Y project, you can focus on folders, symbolics, system icons, etc.

Thanks for listening to the community Clem, have a nice day my friend! :)

**Comment by antrrax** [comment]
I don't know what the profile of Linux Mint users around the world is. In Brazil the vast majority of users are home users, not programmers. And it's usually people who are having initial contacts in the 'Linux world'.

Linux Mint is a very good system, easy to use, very friendly, it is fast with old hardware. But really, for this "domestic" audience; computer laymen (who had only contact with Windows in their lives), the biggest impact is the Linux Mint icon theme. Current icons visually do not represent their applications. As it was said if asking the person to identify which program is just looking at the icon, they will have difficulty recognizing what it is about.

The changes that have happened in the visuals lately have been very good, New default wallpaper has become Beautiful. The Grub Menu is also excellent. The new menu logo was very pleasant. The changes came and were good ...

Changing only to the original icon, I also disagree, would lose the visual consistency of the entire system.

But the proposal to use a template with the original program icon inserted into that template seems like a good idea.

A suggestion that I don't know if it is viable in terms of programming and the time spent doing it, would be to create the icon at the time of installation:
Joining template with standard application icon and using an algorithm that could capture in the application icon the best background color to use in the template. And that suits the Light and Dark theme

**Comment by antrrax** [comment]
This lack of visual identity in the program icons generates another problem in the theme of Mint Y icons, is that some applications share the same icon.

For example:
gnome-screenshot --- and --- simplescreenrecorder have the same icon and make it difficult to recognize these programs if both are fixed to the taskbar or the favorites bar

This must also happen like other programs.
<a href="https://ibb.co/QpjPtqq"><img src="https://i.ibb.co/k564vrr/icons.png" alt="icons" border="0"></a>

**Comment by Diolinux** [comment]
@antrrax yes, this is another question very important, thanks to bring this to light. If Mint ditches the custom icons for programs only, this would be solved. Add some transparency to the panel and menu e move the BIG "show desktop' icon to the corner would be nice too. @clefebvre hope you think about it, we had year of content and Linux Mint support in our community and we hear about this kind of thing every day. Another thing is the workspaces, It is very interesting to have an applet do indicates the existence of those, cause they are active by default, but if the people don't know the shortcut, they are basically useless.
